File: Domain/Product.cs

namespace Domain
{
    using System.ComponentModel.DataAnnotations.Schema;


    public class Product
    {
        public Guid Id { get; set; }  // Renamed to match standard convention
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string Type { get; set; }
        public string Brand { get; set; }
        public int QuantityInStock { get; set; }
        public bool IsSuccess { get; set; }

        [NotMapped]
        public object Value { get; set; } // Not stored in the database
    }
}

File: client-app/src/app/models/product.ts
export interface Product {
    id: string; // Unique identifier for the product
    name: string; // Name of the product
    description: string; // Description of the product
    price: number; // Price of the product
    pictureUrl: string; // URL of the product's image
    type: string; // Type/category of the product
    brand: string; // Brand of the product
    quantityInStock: number; // Quantity available in stock
    isSuccess: boolean; // Indicates if the product operation is successful
    value?: any; // Optional property not mapped to the database
  }